libs/feature/dataviz/src/lib/service/data.service.ts

```diff
@@ -156,20 +156,25 @@ export class DataService {
   async getDownloadLinksFromOgcApiFeatures(
     ogcApiLink: DatasetServiceDistribution
   ): Promise<DatasetDistribution[]> {
-    const collectionInfo = await this.getDownloadUrlsFromOgcApi(
-      ogcApiLink.url.href
-    )
-
-    return Object.keys(collectionInfo.bulkDownloadLinks).map((downloadLink) => {
-      return {
-        ...ogcApiLink,
-        type: 'download',
-        url: new URL(collectionInfo.bulkDownloadLinks[downloadLink]),
-        mimeType: getMimeTypeForFormat(
-          getFileFormatFromServiceOutput(downloadLink)
-        ),
-      }
-    })
+    try {
+      const collectionInfo = await this.getDownloadUrlsFromOgcApi(
+        ogcApiLink.url.href
+      )
+      return Object.keys(collectionInfo.bulkDownloadLinks).map(
+        (downloadLink) => {
+          return {
+            ...ogcApiLink,
+            type: 'download',
+            url: new URL(collectionInfo.bulkDownloadLinks[downloadLink]),
+            mimeType: getMimeTypeForFormat(
+              getFileFormatFromServiceOutput(downloadLink)
+            ),
+          }
+        }
+      )
+    } catch (error) {
+      return Promise.resolve([])
```

Comment:
I think the error can be caught much later, this is already the case for WFS for instance. The errors are caught here: https://github.com/geonetwork/geonetwork-ui/blob/3371b676033bae088ee0423b96863a4eed463e2a/apps/datahub/src/app/record/record-downloads/record-downloads.component.ts#L78

What's happening is maybe that
1/ even if an error is caught it is not shown (could be shown using an overlay for instance?)
2/ when an error is caught, it is assumed that the error comes from wfs; ut now, it can come from somewhere else, so the logic has to be adjusted

Reply:
I think this looks better now, we could probably have a common `handleError` callback which assigns the error and returns `of([])` and use that both for WFS and OGC API. I'm not sure using a Promise is needed here anyway? But yes to have the equivalent of `of` you probably need to do `from(Promise.resolve(...))`, that makes sense to me.

Reply:
Hum, I was not able to refactor this further with a common `handleError` function. Any attempt I did to try to catch the error once the `Promise` has been transformed into en `Obervable` broke the download list, so I needed to keep catching it within the `Promise`.
I've added a separate translation key for the OGC API error. If I saw right, `ogc-client` does not provide more info on the error as for `WFS`, so there is just one generic message so far. It is displayed in the preview components like the map, but I decided to not also add it to the download list to not "pollute" the UI when there might be many links already displayed from working sources.